dipy/reconst/ivim.py

```diff
@@ -0,0 +1,383 @@
+            jac=None, bounds=((0, None), (0, 1.), (0, 1.), (0, 1.)), tol=1e-25, algorithm='L-BFGS-B',
+            gtol=1e-25, ftol=1e-25, eps=1e-15, maxiter=1000):
+        """ Fit method of the Ivim model class
+
+        Parameters
+        ----------
+        data : array
+            The measured signal from one voxel.
+        mask : array
+            A boolean array used to mark the coordinates in the data that
+            should be analyzed that has the shape data.shape[:-1]
+        x0 : array
+            Initial guesses for the parameters S0, f, D_star and D
+            Default x0 = [1.0, 0.10, 0.001, 0.0009]
+            These guess parameters are taken from the Federau paper
+            Dimension can either be 1 or (N, 4) where N is the number of
+            voxels in the data.
+        fit_method: str
+            Use one-stage fitting or two-stage fitting.
+        jac : Boolean
+            Use the Jacobian
+        bounds : tuple
+            Bounds for the various parameters
+        tol : float
+            tolerance for convergence
+        method : str
+            Fitting algorithm to use from scipy.optimize
+
+        Returns
+        -------
+        IvimFit object
+        """
+        if mask is None:
+            # Flatten it to 2D either way:
+            data_in_mask = np.reshape(data, (-1, data.shape[-1]))
+        else:
+            # Check for valid shape of the mask
+            if mask.shape != data.shape[:-1]:
+                raise ValueError("Mask is not the same shape as data.")
+            mask = np.array(mask, dtype=bool, copy=False)
+            data_in_mask = np.reshape(data[mask], (-1, data.shape[-1]))
+
+        if self.min_signal is None:
+            min_signal = _min_positive_signal(data)
+        else:
+            min_signal = self.min_signal
+
+        # Generate guess_parameters for all voxels
+        if x0 is None:
+            x0 = np.ones(
+                (data.shape[:-1] + (4,))) * [1.0, 0.10, 0.001, 0.0009]
+        else:
+            if x0.shape != (data.shape[0], 4):
+                raise ValueError(
+                    "Guess params should be of shape (num_voxels,4)")
+        data_in_mask = np.maximum(data_in_mask, min_signal)
+        x0 = np.reshape(x0[mask], (-1, x0.shape[-1]))
+
+        if fit_method == "one_stage":
+            params_in_mask = one_stage(data_in_mask, self.gtab, x0,
+                                       jac, bounds, tol,
+                                       routine,
+                                       algorithm, gtol, ftol, eps, maxiter)
+        elif fit_method == "two_stage":
+            params_in_mask = two_stage(data_in_mask, self.gtab, x0,
+                                       self.split_b, jac, bounds, tol,
+                                       routine,
+                                       algorithm, gtol, ftol, eps, maxiter)
+        else:
+            raise ValueError("""Fit method must be either
```

Comment:
Please don't break the error string into two lines. Something like this should work: 

    else:
        e_s = "Fit method must be either "
        e_s += "'one_stage' or 'two_stage'"
        ValueError(e_s)